plugins/wasm-cpp/extensions/jwt_auth/README_EN.md

```diff
@@ -0,0 +1,399 @@
+```
+
+**JWT is set in HTTP request header**
+```bash
+curl  http://xxx.hello.com/test -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjEyMyJ9.eyJpc3MiOiJhYmNkIiwic3ViIjoidGVzdCIsImlhdCI6MTY2NTY2MDUyNywiZXhwIjoxODY1NjczODE5fQ.-vBSV0bKeDwQcuS6eeSZN9dLTUnSnZVk8eVCXdooCQ4'
+```
+
+After authentication and authorization, a `X-Mse-Consumer` field will be added in the request header with a value of `consumer1` in this example, to identify the name of the caller.
+
+#### The following requests will be denied:
+
+**Request without JWT provided, returns 401**
+```bash
+curl  http://xxx.hello.com/test
+```
+
+**The consumer matched by the provided JWT in the request does not have access, returns 403**
+```bash
+# consumer1 is not in the allow list of *.example.com
+curl  'http://xxx.example.com/test' -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjEyMyJ9.eyJpc3MiOiJhYmNkIiwic3ViIjoidGVzdCIsImlhdCI6MTY2NTY2MDUyNywiZXhwIjoxODY1NjczODE5fQ.-vBSV0bKeDwQcuS6eeSZN9dLTUnSnZVk8eVCXdooCQ4'
+```
+
+### Enabling at Gateway Instance Level
+
+The following configuration does not specify the `_rules_` field, so JWT authentication will be enabled at the gateway instance level:
+
+```yaml
+consumers:
+- name: consumer1
+  issuer: abcd
+  jwks: |
+    {
+      "keys": [
+        {
+          "kty": "oct",
+          "kid": "123",
+          "k": "hM0k3AbXBPpKOGg__Ql2Obcq7s60myWDpbHXzgKUQdYo7YCRp0gUqkCnbGSvZ2rGEl4YFkKqIqW7mTHdj-bcqXpNr-NOznEyMpVPOIlqG_NWVC3dydBgcsIZIdD-MR2AQceEaxriPA_VmiUCwfwL2Bhs6_i7eolXoY11EapLQtutz0BV6ZxQQ4dYUmct--7PLNb4BWJyQeWu0QfbIthnvhYllyl2dgeLTEJT58wzFz5HeNMNz8ohY5K0XaKAe5cepryqoXLhA-V-O1OjSG8lCNdKS09OY6O0fkyweKEtuDfien5tHHSsHXoAxYEHPFcSRL4bFPLZ0orTt1_4zpyfew",
+          "alg": "HS256"
+        }
+      ]
+    }
+- name: consumer2
+  issuer: abc
+  jwks: |
+    {
+      "keys": [
+        {
+          "kty": "RSA",
+          "e": "AQAB",
+          "use": "sig",
+          "kid": "123",
+          "alg": "RS256",
+          "n": "i0B67f1jggT9QJlZ_8QL9QQ56LfurrqDhpuu8BxtVcfxrYmaXaCtqTn7OfCuca7cGHdrJIjq99rz890NmYFZuvhaZ-LMt2iyiSb9LZJAeJmHf7ecguXS_-4x3hvbsrgUDi9tlg7xxbqGYcrco3anmalAFxsbswtu2PAXLtTnUo6aYwZsWA6ksq4FL3-anPNL5oZUgIp3HGyhhLTLdlQcC83jzxbguOim-0OEz-N4fniTYRivK7MlibHKrJfO3xa_6whBS07HW4Ydc37ZN3Rx9Ov3ZyV0idFblU519nUdqp_inXj1eEpynlxH60Ys_aTU2POGZh_25KXGdF_ZC_MSRw"
+        }
+      ]
+    }
+```
+
+# Common Error Codes
+| 
+HTTP Status Code | Error Message               | Reason Description|
+| ----------- | ---------------------- | -------------------------------------------------------------------------------- |
+| 401         | Jwt missing            | The JWT is not provided in the request header. |
+| 401         | Jwt expired            | The JWT has expired. |
+| 401         | Jwt verification fails | The JWT payload verification failed, such as when the iss claim does not match. |
```

Comment:
`such as when`, I don't think it is right from the grammar standpoint, although we can live with it.